src/neptune/internal/operation_processors/operation_storage.py

```diff
@@ -34,11 +34,12 @@ def __init__(self, data_path: Union[str, Path]):
             self._data_path = data_path
         else:
             self._data_path = Path(data_path)
-
         # initialize directories
         os.makedirs(self.data_path, exist_ok=True)
```

Comment:
I think `os.makedirs` accepts both `str` and `Path` so this could be even before the conversion and first assignment of `self_data_path`.

Reply:
So let's say I have
```
def __init__(self, data_path: Union[str, Path]):
    os.makedirs(data_path, exist_ok=True)
    
    self._data_path = Path(data_path).resolve()
    
    os.makedirs(self.upload_path, exist_ok=True)

@property
def data_path(self) -> Path:
    return self._data_path

@property
def upload_path(self) -> Path:
    return self.data_path / "upload_path"
```

> `os.makedirs(self.upload_path, exist_ok=True)`
will result in PermissionError, as now the upload path will be absolute.